host/xtest/stats.c

```diff
@@ -209,6 +218,99 @@ static int stat_memleak(int argc, char *argv[])
 	return close_sess(&ctx, &sess);
 }
 
+static int stat_loaded_ta(int argc, char *argv[])
+{
+	TEEC_Context ctx = { };
+	TEEC_Session sess = { };
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	uint32_t eo = 0;
+	TEEC_Operation op = { };
+	void *buff = NULL;
+	struct ta_dump_stats *stats = NULL;
+	size_t stats_size_bytes = 0;
+	size_t n = 0;
+	uint32_t retry_count = 10;
+
+	UNUSED(argv);
+	if (argc != 1)
+		return usage();
+
+	open_sess(&ctx, &sess);
+retry:
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
+	res = TEEC_InvokeCommand(&sess, STATS_CMD_TA_STATS, &op, &eo);
+	if (res != TEEC_ERROR_SHORT_BUFFER)
+		errx(EXIT_FAILURE,
+		     "TEEC_InvokeCommand: res %#"PRIx32" err_orig %#"PRIx32,
+		     res, eo);
+
+	stats_size_bytes = op.params[0].tmpref.size;
+	if (stats_size_bytes == 0)
+		errx(EXIT_FAILURE, "No loaded TA found");
```

Comment:
This is not an error case. `errx()` is dedicated to error case so prefer use here `return close_sess(&ctx, &sess);` as you initially implemented.

Reply:
okay, done.